projects/baby_driver/test/test_spi_exchange.c

```diff
@@ -0,0 +1,354 @@
+#include "spi_exchange.h"
+
+#include <string.h>
+
+#include "babydriver_msg_defs.h"
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "controller_board_pins.h"
+#include "dispatcher.h"
+#include "gpio_mcu.h"
+#include "log.h"
+#include "ms_test_helper_can.h"
+#include "ms_test_helpers.h"
+#include "spi.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define INVALID_CS_PORT (NUM_GPIO_PORTS)
+#define INVALID_CS_PIN (GPIO_PINS_PER_PORT)
+
+#define INVALID_SPI_PORT (2)
+#define INVALID_SPI_MODE (4)
+
+typedef enum {
+  TEST_CAN_EVENT_TX = 0,
+  TEST_CAN_EVENT_RX,
+  TEST_CAN_EVENT_FAULT,
+  NUM_TEST_CAN_EVENTS,
+} TestCanEvent;
+
+static CanStorage s_can_storage;
+
+static uint8_t s_callback_counter_status;
+static uint8_t s_num_messages;
+static uint8_t s_received_data_status[8];
```

Comment:
You don't really need to store all 8 bytes of the status message, eh? You already know that the 0th byte is `BABYDRIVER_MESSAGE_STATUS` (since the callback wouldn't be called otherwise!), so you don't need to test it. You only really need to store the status in byte 1.